### PR TITLE
SAK-49560 LTI Create a single "Add Learning App" UI to be used across Sakai

### PR DESCRIPTION
**Pull request by csev**

### CODE REVIEW

File: basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/BaseLTIService.java [hunk 1]
```diff
@@ -573,7 +573,7 @@ public List<Map<String, Object>> getToolsLaunch(String siteId) {
 	public List<Map<String, Object>> getToolsLaunchCourseNav(String siteId, boolean includeStealthed) {
 		String query = "( lti_tools."+LTIService.LTI_MT_LAUNCH+" = 1 AND " +
 			"lti_tools."+LTIService.LTI_PL_COURSENAV+" = 1 )";
-		return getTools(query, null, 0, 0, siteId, includeStealthed);
+		return getTools(query, "title", 0, 0, siteId, includeStealthed);
```
Comment (jesusmmp) [inline basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/BaseLTIService.java:576]: Put "title" as a constant

Comment (csev) [reply basiclti/basiclti-impl/src/java/org/sakaiproject/lti/impl/BaseLTIService.java:576]: Good catch and easy fix.   Thanks!

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 2]
```diff
@@ -226,6 +226,12 @@ public String buildErrorPanelContext(VelocityPortlet portlet, Context context,
 			RunData rundata, SessionState state) {
 		context.put("tlang", rb);
 		context.put("includeLatestJQuery", PortalUtils.includeLatestJQuery("LTIAdminTool"));
+
+		String alertMessage = (String) state.getAttribute(STATE_MESSAGE);
+		state.removeAttribute(STATE_MESSAGE);
+		log.error("alertMessage = {}", alertMessage);
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:232]:
```suggestion
		log.warn("alertMessage = {}", alertMessage);
```

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 3]
```diff
@@ -2097,10 +2108,28 @@ public void doMultipleContentItemResponse(RunData data, Context context) {
...
+			log.debug("Lessons flow, redirecting to returnUrl {}", returnUrl);
+			state.setAttribute(STATE_REDIRECT_URL, returnUrl);
+			forward = "Redirect";
+		} else if ( flow.equals(FLOW_PARAMETER_ASSIGNMENT) ) {
 			forward = "AssignmentDone";
-		} else {
+		} else if ( flow.equals(FLOW_PARAMETER_EDITOR) ) {
 			forward = "CKEditorDone";
+		} else {
+			log.error("Unhandled flow type {}", flow);
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:2129]:
```suggestion
			log.warn("Unhandled flow type {}", flow);
```

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/BltiPickerProducer.java [hunk 4]
```diff
@@ -101,21 +119,19 @@ public void fillComponents(UIContainer tofill, ViewParameters viewparams, Compon
 		    try {
 			simplePageBean.updatePageObject(((GeneralViewParameters) viewparams).getSendingPage());
 		    } catch (Exception e) {
-			log.info("QuizPicker permission exception " + e);
+			log.info("BltiPicker permission exception " + e);
```
Comment (ern) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/BltiPickerProducer.java:122]:
```suggestion
			log.warn("BltiPicker permission exception, {}", e.toString());
```

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 5]
```diff
@@ -2097,10 +2108,28 @@ public void doMultipleContentItemResponse(RunData data, Context context) {
 		}
 
 		String forward;
-		if ( flow.equals(FLOW_PARAMETER_ASSIGNMENT) ) {
+		if ( flow.equals(FLOW_PARAMETER_LESSONS) ) {
+			if  (contentKey == null ) {
+				log.error("Returning content item to Lessons, but contentKey={}", contentKey);
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:2113]:
```suggestion
				log.warn("Returning content item to Lessons, but contentKey={}", contentKey);
```

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 6]
```diff
@@ -2866,14 +2919,29 @@ public String buildPostContentConfigPanelContext(VelocityPortlet portlet, Contex
 				context.put("toolTitle", (String) content.get(LTIService.LTI_TITLE));
 			}
 			return "lti_assignment_return";
-		}
+		} else if ( flow.equals(FLOW_PARAMETER_LESSONS) ) {
+			if  (contentKey == null ) {
+				log.error("Returning content item to Lessons, but contentKey={}", contentKey);
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:2924]:
```suggestion
				log.warn("Returning content item to Lessons, but contentKey={}", contentKey);
```

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 7]
```diff
@@ -2866,14 +2919,29 @@ public String buildPostContentConfigPanelContext(VelocityPortlet portlet, Contex
...
-		// TODO: Someday handle non CI/DL FLOW_PARAMETER_LESSONS
+			log.debug("Lessons flow, redirecting to returnUrl {}", returnUrl);
+			context.put("returnUrl", returnUrl);
+			return "lti_content_redirect";
+		} else if ( flow.equals(FLOW_PARAMETER_EDITOR) ) {
+			context.put("new_content", new_content);
+			context.put("goodcount", new Integer(1));
+			return "lti_editor_done";
+		} else {
+			log.error("Unhandled flow type {}", flow);
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:2940]:
```suggestion
			log.warn("Unhandled flow type {}", flow);
```

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html [hunk 8]
```diff
@@ -146,11 +76,17 @@ <h5 class="modal-title" id="bltiModalLabel">Modal title</h5>
       });
 
       el.addEventListener("hidden.bs.modal", e => {
-        if ( doreload ) {
-          location.reload();
-        } else {
-          $('#sakai-basiclti-admin-iframe').attr('src','/library/image/sakai/spinner.gif');
-        }
+          console.log("hidden.bs.modal");
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/templates/BltiPicker.html:79]: should this be console.debug?

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html [hunk 9]
```diff
@@ -162,26 +98,27 @@ <h5 class="modal-title" id="bltiModalLabel">Modal title</h5>
...
-//]]>
 
-//If none of the tools have a description, hides the column.
 $( document ).ready(function() {
-    var emptyDescriptionCount = $('.externalToolContainerClass td:nth-child(2):not(:has(:not(:empty)))').length;
-    var rowNumber = $('.externalToolContainerClass tbody tr').length;
-    if(emptyDescriptionCount == rowNumber){
-        $('.externalToolContainerClass td:nth-child(2), .externalToolContainerClass th:nth-child(2)').hide();
-    }
+    console.log("READY");
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/templates/BltiPicker.html:103]: should this be console.debug

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html [hunk 10]
```diff
@@ -162,26 +98,27 @@ <h5 class="modal-title" id="bltiModalLabel">Modal title</h5>
...
+		// If we have the blti-autosubmit form, submit it therwise show picker
+		setTimeout(function(){
+			var form = document.getElementById("blti-autosubmit");
+            if ( form != null ) {
+                form.submit();
+            } else {
+                console.log("There is no form to submit");
```
Comment (ern) [inline lessonbuilder/tool/src/webapp/templates/BltiPicker.html:110]: console.debug?

Comment (csev) [reply lessonbuilder/tool/src/webapp/templates/BltiPicker.html:110]: No - this is a really profound screw up.  It means things are very broken.  What will happen next is failure so we might as well have a serious log message.

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html (hunk 8 above)
Comment (csev) [reply lessonbuilder/tool/src/webapp/templates/BltiPicker.html:79]: Yes

File: lessonbuilder/tool/src/webapp/templates/BltiPicker.html (hunk 9 above)
Comment (csev) [reply lessonbuilder/tool/src/webapp/templates/BltiPicker.html:103]: Yes

**Pull request merged**